cmake/Modules/LocalDav1d.cmake

```diff
@@ -0,0 +1,106 @@
+if(NOT AVIF_LOCAL_DAV1D_TAG)
+    set(AVIF_LOCAL_DAV1D_TAG "1.2.1")
+endif()
+if(NOT AVIF_LOCAL_DAV1D_REPO)
+    set(AVIF_LOCAL_DAV1D_REPO "https://code.videolan.org/videolan/dav1d.git")
+endif()
+
+function(avif_build_local_dav1d)
+    set(source_dir "${AVIF_SOURCE_DIR}/ext/dav1d")
```

Comment:
In this PR I opted to put the source dirs in the same place, but the default behavior is for the source to also be in the build directory. The rules for determining the directories are in the cmake docs at the end of [this section](https://cmake.org/cmake/help/latest/module/ExternalProject.html#directory-options). Is there a preference for these directory options?

Reply:
I'd say the only reason to keep it in ext would be for legacy. ext makes it clearer to:
- tweak build parameters (but now that will be in the Local*cmake files)
- patch the repos
If we modify the source of a FetchContent, do they get reset anyway by CMake? Or is the only way to patch the sources through the CMake file?

Reply:
There are a few ways to prevent cmake from overwriting local changes to source repos. One option is to set `FETCHCONTENT_SOURCE_DIR_<PackageName>`; it allows you to override the source directory, and FetchContent will use whatever is there instead of fetching it. You could also make `DOWNLOAD_COMMAND` a no-op `if(EXISTS)`, either by default or based on a user option. If we ended up going that route maybe we would only put SOURCE_DIR in ext if that flag is present.

Reply:
Ok, so as long as it is documented, I think we do not care. My preference would be to check if there is anything in ext: if so, take it, otherwise FetchContent it in build.
A big use case we have is to find bugs in libaom/dav1d: we might need to go down those repos and modify them and/or use a specific tag.